rk4_integrator/plugin.cpp

```diff
@@ -159,39 +167,39 @@ class imu_integrator : public plugin {
     }
 
 	// Select IMU readings based on timestamp similar to how OpenVINS selects IMU values to propagate
-	std::vector<imu_type> select_imu_readings(const std::vector<imu_type>& imu_data, double time_begin, double time_end) {
+	std::vector<imu_type> select_imu_readings(const std::vector<imu_type>& imu_data, time_point time_begin, time_point time_end) {
 		std::vector<imu_type> prop_data;
 		if (imu_data.size() < 2) {
 			return prop_data;
 		}
 
-		for (unsigned i = 0; i < imu_data.size()-1; i++) {
+		for (int i = 0; i < int(imu_data.size())-1; i++) {
 
 			// If time_begin comes inbetween two IMUs (A and B), interpolate A forward to time_begin
-			if (imu_data.at(i+1).timestamp > time_begin && imu_data.at(i).timestamp < time_begin) {
-				imu_type data = interpolate_imu(imu_data.at(i), imu_data.at(i+1), time_begin);
+			if (imu_data[i+1].timestamp > time_begin && imu_data[i].timestamp < time_begin) {
+				imu_type data = interpolate_imu(imu_data[i], imu_data[i+1], time_begin);
 				prop_data.push_back(data);
 				continue;
 			}
 
 			// IMU is within time_begin and time_end
-			if (imu_data.at(i).timestamp >= time_begin && imu_data.at(i+1).timestamp <= time_end) {
-				prop_data.push_back(imu_data.at(i));
+			if (imu_data[i].timestamp >= time_begin && imu_data[i+1].timestamp <= time_end) {
+				prop_data.push_back(imu_data[i]);
 				continue;
 			}
 
 			// IMU is past time_end
-			if (imu_data.at(i+1).timestamp > time_end) {
-				imu_type data = interpolate_imu(imu_data.at(i), imu_data.at(i+1), time_end);
+			if (imu_data[i+1].timestamp > time_end) {
+				imu_type data = interpolate_imu(imu_data[i], imu_data[i+1], time_end);
 				prop_data.push_back(data);
 				break;
 			}
 		}
 
 		// Loop through and ensure we do not have an zero dt values
 		// This would cause the noise covariance to be Infinity
-		for (size_t i = 0; i < prop_data.size()-1; i++) {
-			if (std::abs(prop_data.at(i+1).timestamp-prop_data.at(i).timestamp) < 1e-12) {
+		for (int i = 0; i < int(prop_data.size())-1; i++) {
+			if (std::chrono::abs(prop_data[i+1].timestamp - prop_data[i].timestamp) < std::chrono::nanoseconds{1}) {
```

Comment:
Reference code was 1 picosecond.

Reply:
Since the timestamps are stored as an integer number of nanoseconds, this makes more sense.